src/window.rs

```diff
@@ -528,18 +528,24 @@ impl Window {
     /// events have been processed by the event loop.
     ///
     /// This is the **strongly encouraged** method of redrawing windows, as it can integrate with
-    /// OS-requested redraws (e.g. when a window gets resized).
+    /// OS-requested redraws (e.g. when a window gets resized). To improve the event delivery
+    /// consider using [`Window::pre_present_notify`] as described in docs.
     ///
     /// This function can cause `RedrawRequested` events to be emitted after [`Event::MainEventsCleared`]
     /// but before `Event::NewEvents` if called in the following circumstances:
     /// * While processing `MainEventsCleared`.
     /// * While processing a `RedrawRequested` event that was sent during `MainEventsCleared` or any
     ///   directly subsequent `RedrawRequested` event.
     ///
+    /// However the event may not arrived on the next event loop, winit will try to align the
+    /// receiving the evente with the windowing system drawing loop.
```

Comment:
I'm not sure what this sentence is trying to say.

Reply:
We have a concept that if you request a redraw before the `MainEventsCleared` you'll get the `RedrawRequsted` on the current loop iteration after it, I'm trying to say that it's no longer guaranteed.

Reply:
Isn't that implicit now that it's bound to the windowing system drawing loop?

Reply:
Well, we need to say that and that's what this sentence is trying to do.

Though, maybe better wording here is required.

Reply:
I think this is the part we ought to change as well:
https://github.com/rust-windowing/winit/blob/ec5adcc83ec6eb12a03582809964fc1e7da28699/src/window.rs#L527-L528

```diff
-    /// Emits a [`Event::RedrawRequested`] event in the associated event loop after all OS
-    /// events have been processed by the event loop.
+    /// Queues a [`Event::RedrawRequested`] event to be emitted that aligns with the windowing
+    /// system drawing loop.
```

```suggestion
    /// However as the event aligns with the windowing system drawing loop, it may not arrive in
    /// the next event loop.
```

English isn't my first language, but I hope that's good enough.